value-annotations/src/org/immutables/value/Value.java

```diff
@@ -809,7 +809,7 @@
     ImplementationVisibility visibility() default ImplementationVisibility.SAME;
 
     /**
-     * Specify whether init and copy methods for an unwrapped {@code X} of {@code Optional<X>}
+     * Specify whether init, copy and factory methods and constructors for an unwrapped {@code X} of {@code Optional<X>}
```

Comment:
I see we're supposed to cover this features under existing feature flag `optionalAcceptNullable`, which is good in my opinion. However, (plz correct if I'm wrong) when we generating code we only ask about `type.hasOptionalConstructorArguments` but we're not querying this `optionalAcceptNullable` style attribute, thus we always generate additional factory methods if we have at least one optional attribute, which is not good

Reply:
I think what I wanted to do was to add a `Objects.requireNonNull()` to the corresponding constructor when the flag was not true. But it seems I totally forgot about that in the end, sorry about that. 
In my opinion this would make more sense as it would behave the same as the generated with (copy) methods.
What do you think, would it be ok if I added the `Objects.requireNonNull()`?